File: src/main/java/giraffe/miausicbox/model/Blog.java
package giraffe.miausicbox.model;

import java.util.Date;
import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Blog {

	public interface BasicAtt {}
	
	// Attributes
	@JsonView(BasicAtt.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(BasicAtt.class)
	private String name;
	
	@JsonView(BasicAtt.class)
	private String image;
	
	@JsonView(BasicAtt.class)
	private String text;
	
	@JsonView(BasicAtt.class)
	private Date date;
	
	// Constructor
	public Blog() {}
	
	public Blog(
			String name,
			String image,
			String text,
			Date date
			) {
		super();
		this.name = name;
		this.image = image;
		this.text = text;
		this.date = date;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}
	
	// ToString
	@Override
	public String toString() {
		return (this.getName() + " (" + this.getDate().toString() + ")");
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Blog other = (Blog) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getName(), other.getName())
				&& (Objects.equals(this.getDate(), other.getDate())));
	}
	
}

File: src/main/resources/static/app/classes/Blog.ts
/**
 *  Class that represent the information related to MiausicBox blogs
 *  @class Blog
 */

export class Blog {

  /* Attributes */
  protected _id:number;
  protected _name:string;
  protected _image:string;
  protected _text:string;
  protected _date:Date;

  /* Constructor */
  constructor(id:number, name:string, image:string, text:string, date:Date) {
    this._id = id;
    this._name = name;
    this._image = image;
    this._text = text;
    this._date = date;
  }

  /* Getters & Setters */
  get id():number {
    return this._id;
  }

  get name():string {
    return this._name;
  }

  get image():string {
    return this._image;
  }

  get text():string {
    return this._text;
  }

  get date():Date {
    return this._date;
  }

  set id(id:number) {
    this._id = id;
  }

  set name(name:string) {
    this._name = name;
  }

  set image(image:string) {
    this._image = image;
  }

  set text(text:string) {
    this._text = text;
  }

  set date(date:Date) {
    this._date = date;
  }

  /** Return if an Blog is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any) {
    if (!(object instanceof Blog)) {
      return false;
    } else {
      var blog:Blog = object;
      return (blog.name == this.name &&
              blog.date == this.date);
    }
  }

  /** Return if an Blog has an image
   *  @method hasImage
   *  @param {} boolean
   */
  hasImage():boolean {
    return (this.image != "");
  }

}
